torchdata/datapipes/map/util/cacheholder.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from typing import Callable, List, Optional, TypeVar
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.map import MapDataPipe
+
+
+T_co = TypeVar("T_co", covariant=True)
+
+
+def list_cache(size: int) -> List:
+    cache = [None] * size
+    try:
+        from numpy import array
+
+        cache = array(cache)
+    except (ImportError, ModuleNotFoundError):
+        pass
+    return cache
```

Comment:
My intention here is to default to a `numpy.array` when available, and a `List` when it is not.

I thought about the memory leak issues related to DataLoader when `List` is used and I want to avoid that by defaulting the `numpy.array` instead. Any thoughts on how this should be handled here?